sdv/metadata/multi_table.py

```diff
@@ -170,6 +170,25 @@ def _validate_foreign_child_key(self, child_table_name, parent_table_name, child
                 'with a non-primary key.'
             )
 
+    def _validate_foreign_key_is_not_reused(
+        self, parent_table_name, parent_primary_key, child_table_name, child_foreign_key
+    ):
+        for relationship in self.relationships:
+            foreign_key_already_used = (
```

Comment:
won't this cause the detection to error if it accidentally tries to add a relationship with a foreign key that's already used? I don't think it should crash. Instead I think the detection should just not detect a foreign key it already has seen

Reply:
nvm the add_relationship is wrapped in a try/catch